diff --git a/launcher/FileSystem.cpp b/launcher/FileSystem.cpp
index 000000000..111111111 100644
--- a/launcher/FileSystem.cpp
+++ b/launcher/FileSystem.cpp
@@ -109,7 +109,8 @@
 
 // Avoid re-defining structs retroactively added to MinGW
 // https://github.com/mingw-w64/mingw-w64/issues/90#issuecomment-2829284729
-#if __MINGW64_VERSION_MAJOR < 13
+#if !defined(_DUPLICATE_EXTENTS_DATA) && \
+    (!defined(__MINGW64_VERSION_MAJOR) || __MINGW64_VERSION_MAJOR < 13)
 
 struct _DUPLICATE_EXTENTS_DATA {
     HANDLE FileHandle;
